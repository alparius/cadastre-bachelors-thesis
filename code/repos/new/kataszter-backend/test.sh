#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
