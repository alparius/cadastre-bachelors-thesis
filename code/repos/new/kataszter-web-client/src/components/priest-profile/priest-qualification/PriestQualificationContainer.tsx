import React from "react";
import { Button, Card, Divider, Header, Icon } from "semantic-ui-react";

import IPriestProfileData from "../../../data/priest/PriestProfileData";
import { NewQualification } from "../../../data/priest/Qualification";
import BoldFM from "../../../util/styledMessages/BoldFM";
import ItalicFM from "../../../util/styledMessages/ItalicFM";
import ErrorMsg from "../../ErrorMsg";
import ProfilePlaceholder from "../../ProfilePlaceholder";
import LanguageCard from "./LanguageCard";
import PriestProfileQualificationList from "./PriestQualificationList";

interface IPriestQualificationContainerProps {
    priest: IPriestProfileData;
    loading: boolean;
    showError: boolean;
    editMode: boolean;
    setPriest(newPriest: IPriestProfileData): void;
}

const PriestQualificationContainer: React.FC<IPriestQualificationContainerProps> = (props: IPriestQualificationContainerProps) => {
    const { priest, setPriest, loading, showError, editMode } = props;

    const handleNewQualification = () => {
        if (priest) {
            if (priest.qualifications) {
                priest.qualifications.splice(0, 0, { ...NewQualification });
                setPriest({
                    ...priest,
                    qualifications: priest.qualifications
                });
            } else {
                setPriest({
                    ...priest,
                    qualifications: [{ ...NewQualification }]
                });
            }
        }
    };

    const handleChangeOnLanguageSkills = (labelName: string, value: any) => {
        setPriest({ ...priest, [labelName]: value });
    };

    if (showError) {
        return <ErrorMsg errorMessage={"error.connection"} />;
    } else if (loading || !priest) {
        return <ProfilePlaceholder />;
    } else {
        return (
            <>
                <Header as="h2">
                    <ItalicFM id="profile.languageSkills" />
                </Header>
                <Card.Group stackable doubling itemsPerRow={2}>
                    <LanguageCard
                        label="speakingSkills"
                        languageSkills={priest.speakingSkills}
                        editMode={editMode}
                        handleChangeOnLanguageSkills={handleChangeOnLanguageSkills}
                    />
                    <LanguageCard
                        label="writingSkills"
                        languageSkills={priest.writingSkills}
                        editMode={editMode}
                        handleChangeOnLanguageSkills={handleChangeOnLanguageSkills}
                    />
                </Card.Group>

                <Divider />

                <Header as="h2">
                    {priest.qualifications && priest.qualifications.length > 1 ? (
                        <ItalicFM id="profile.qualifications" />
                    ) : (
                        <ItalicFM id="profile.qualification" />
                    )}
                </Header>
                {editMode && (
                    <Button id="addNew" icon basic color="violet" circular labelPosition="left" onClick={handleNewQualification}>
                        <Icon name="add" color="violet" size="large" />
                        <BoldFM id="profile.addNewQualification" />
                    </Button>
                )}
                <PriestProfileQualificationList priest={priest} editMode={editMode} setPriest={setPriest} />
            </>
        );
    }
};

export default PriestQualificationContainer;
