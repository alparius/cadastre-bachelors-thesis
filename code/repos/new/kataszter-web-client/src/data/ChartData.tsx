interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
