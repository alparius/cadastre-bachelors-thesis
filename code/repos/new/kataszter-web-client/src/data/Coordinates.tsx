interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
