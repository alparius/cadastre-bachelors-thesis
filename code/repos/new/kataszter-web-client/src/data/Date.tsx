interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
