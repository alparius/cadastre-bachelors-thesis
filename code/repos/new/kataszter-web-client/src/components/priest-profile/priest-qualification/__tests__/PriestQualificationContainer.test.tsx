/*===================
  PriestQualificationContainer.unit.test.tsx
 ===================*/
import React from "react";

import IPriestProfileData from "../../../../data/priest/PriestProfileData";
import { NewTestPriestProfileData } from "../../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter, mountWithIntl } from "../../../../util/test/test-wrapper";
import PriestQualificationContainer from "../PriestQualificationContainer";

const handlers = {
    setPriest: jest.fn(),
    setEditMode: jest.fn()
};

const priest: IPriestProfileData = NewTestPriestProfileData;

describe("PriestQualificationContainer element", () => {
    afterEach(() => {
        handlers.setPriest.mockClear();
        handlers.setEditMode.mockClear();
    });

    it("should render a qualification container when the priest is not empty", () => {
        const container = createComponentWithIntlAndRouter(
            <PriestQualificationContainer priest={priest} loading={false} showError={false} editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a placeholder when the priest is empty", () => {
        const emptyPriestFamily: IPriestProfileData[] = [];
        const container = createComponentWithIntlAndRouter(
            <PriestQualificationContainer priest={emptyPriestFamily[0]} loading={false} showError={false} editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message when showError is true", () => {
        const container = createComponentWithIntlAndRouter(
            <PriestQualificationContainer priest={priest} loading={false} showError editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message when showError and loading are both true", () => {
        const container = createComponentWithIntlAndRouter(
            <PriestQualificationContainer priest={priest} loading showError editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a placeholder card when only loading is true", () => {
        const container = createComponentWithIntlAndRouter(
            <PriestQualificationContainer priest={priest} loading showError={false} editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call setPriest if add new Button pressed", () => {
        const container = mountWithIntl(<PriestQualificationContainer priest={priest} loading={false} showError={false} editMode {...handlers} />);
        container.find("#addNew").at(0).simulate("click");
        expect(handlers.setPriest).toHaveBeenCalledTimes(1);
    });
});
