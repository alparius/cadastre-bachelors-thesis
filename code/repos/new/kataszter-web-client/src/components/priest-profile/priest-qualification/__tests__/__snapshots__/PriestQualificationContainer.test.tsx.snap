// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriestQualificationContainer element should render a placeholder card when only loading is true 1`] = `
Array [
  <div
    className="ui four column grid"
  >
    <div
      className="column"
    />
    <div
      className="column"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="rectangular image"
        />
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
      </div>
    </div>
    <div
      className="column"
    />
  </div>,
  <div
    className="ui divider"
  />,
  <div
    className="ui four column grid"
  >
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
  </div>,
]
`;

exports[`PriestQualificationContainer element should render a placeholder when the priest is empty 1`] = `
Array [
  <div
    className="ui four column grid"
  >
    <div
      className="column"
    />
    <div
      className="column"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="rectangular image"
        />
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
      </div>
    </div>
    <div
      className="column"
    />
  </div>,
  <div
    className="ui divider"
  />,
  <div
    className="ui four column grid"
  >
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
  </div>,
]
`;

exports[`PriestQualificationContainer element should render a qualification container when the priest is not empty 1`] = `
Array [
  <h2
    className="ui header"
  >
    <i>
      Language skills
    </i>
  </h2>,
  <div
    className="ui doubling stackable two cards"
  >
    <div
      className="ui fluid card"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="content"
        >
          <div
            className="header"
            id="header"
          >
            <h2
              className="ui header"
            >
              Speaking skills
            </h2>
          </div>
        </div>
        <div
          className="description"
        >
          <div
            className="ui list"
            role="list"
          >
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Well
                    :
                  </b>
                   
                  magyar
                </div>
              </form>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Medium
                    :
                  </b>
                   
                  román
                </div>
              </form>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Less
                    :
                  </b>
                   
                  német
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ui fluid card"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="content"
        >
          <div
            className="header"
            id="header"
          >
            <h2
              className="ui header"
            >
              Writing skills
            </h2>
          </div>
        </div>
        <div
          className="description"
        >
          <div
            className="ui list"
            role="list"
          >
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Well
                    :
                  </b>
                   
                  magyar
                </div>
              </form>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Medium
                    :
                  </b>
                   
                  román
                </div>
              </form>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Less
                    :
                  </b>
                   
                  német
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ui divider"
  />,
  <h2
    className="ui header"
  >
    <i>
      Qualification
    </i>
  </h2>,
  <div
    className="ui doubling stackable two cards"
  >
    <div
      className="ui fluid card"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="content"
        >
          <div
            className="header"
            id="header"
          >
            <h2
              className="ui header"
            >
              Erettsegi bizonyitvany
            </h2>
          </div>
        </div>
        <div
          className="description"
        >
          <div
            className="ui list"
            role="list"
          >
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Genesis date
                    :
                  </b>
                   
                  <span
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    2000.06.06
                  </span>
                </div>
              </form>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Issuing Authority
                    :
                  </b>
                   
                  Felso Oktatasi Intezmeny
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`PriestQualificationContainer element should render an error message when showError and loading are both true 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    Could not estabilish server connection.
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;

exports[`PriestQualificationContainer element should render an error message when showError is true 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    Could not estabilish server connection.
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;
